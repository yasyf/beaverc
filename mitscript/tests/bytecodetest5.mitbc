function
{
	functions =
	[
		function
		{
			functions = [],
			constants = [None],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
				load_const	0
				return
			]
		},
		function
		{
			functions = [],
			constants = [None],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
				load_const	0
				return
			]
		},
		function
		{
			functions = [],
			constants = [None],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
				load_const	0
				return
			]
		},
		function
		{
			functions =
			[
				function
				{
					functions = [],
					constants = ["Rect(", ", ", ")", 1, None],
					parameter_count = 0,
					local_vars = [],
					local_ref_vars = [],
					free_vars = [this, x, y],
					names = [print, x, y],
					instructions = 
					[
						load_const	0
						push_ref	0
						load_ref
						field_load	1
						add
						load_const	1
						add
						push_ref	0
						load_ref
						field_load	2
						add
						load_const	2
						add
						load_const	3
						load_global	0
						call
						pop
						load_const	4
						return
					]
				}
			],
			constants = [3],
			parameter_count = 4,
			local_vars = [x, y, a, b, this],
			local_ref_vars = [this, x, y],
			free_vars = [],
			names = [x, y, a, b, print],
			instructions = 
			[
				alloc_record
				dup
				load_local	0
				field_store	0
				dup
				load_local	1
				field_store	1
				dup
				load_local	2
				field_store	2
				dup
				load_local	3
				field_store	3
				dup
				push_ref	0
				push_ref	1
				push_ref	2
				load_const	0
				load_func	0
				alloc_closure
				field_store	4
				push_ref	0
				swap
				store_ref
				push_ref	0
				load_ref
				return
			]
		}
	],
	constants = [1, 4, 5, 0, None, "yep", "nes", "ted"],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [print, input, intcast, Rect, p, derp, herp, nested, still, yep, still_nested, mega_nested],
	instructions = 
	[
		load_func	0
		store_global	0
		load_func	1
		store_global	1
		load_func	2
		store_global	2
		load_func	3
		store_global	3
		load_const	0
		load_const	1
		load_const	2
		load_const	2
		load_const	1
		load_global	3
		call
		store_global	4
		load_const	3
		load_global	4
		field_load	0
		call
		pop
		load_global	4
		load_const	0
		load_global	0
		call
		pop
		alloc_record
		dup
		alloc_record
		dup
		alloc_record
		dup
		alloc_record
		dup
		alloc_record
		dup
		alloc_record
		dup
		load_const	4
		field_store	5
		field_store	6
		field_store	7
		field_store	8
		field_store	9
		field_store	10
		store_global	11
		load_global	11
		load_global	11
		field_load	10
		load_const	5
		index_load
		field_load	8
		load_const	6
		load_const	7
		add
		index_load
		field_load	6
		swap
		field_store	5
		load_global	11
		field_load	10
		field_load	9
		field_load	8
		field_load	7
		field_load	6
		field_load	5
		load_global	11
		eq
		load_const	0
		load_global	0
		call
		pop
		load_const	3
		return
	]
}