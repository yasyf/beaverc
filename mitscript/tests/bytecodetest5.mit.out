Rect(5, 5)
{print:FUNCTION x:5 y:5 a:4 b:1 }
True
